src/planner/join_order_enumerator.cpp

```diff
@@ -152,7 +152,7 @@ void JoinOrderEnumerator::planNodeScan(uint32_t nodePos) {
 }
 
 static std::pair<std::shared_ptr<NodeExpression>, std::shared_ptr<NodeExpression>>
-getBoundAndNbrNodes(const RelExpression& rel, RelDirection direction) {
+getBoundAndNbrNodes(const RelExpression& rel, RelDataDirection direction) {
```

Comment:
In planner always use ExtendDirection. Change across this file.

You probably need to change the following code block as well
```
for (auto direction : REL_DIRECTIONS) {
        auto plan = std::make_unique<LogicalPlan>();
        auto [boundNode, _] = getBoundAndNbrNodes(*rel, direction);
        appendScanNodeID(boundNode, *plan);
        appendExtendAndFilter(rel, direction, predicates, *plan);
        context->addPlan(newSubgraph, std::move(plan));
    }
```
to enumerate a plan that goes from src to dst and another one that goes from dst to src. Something along the line of
```
// plan src to dst
auto plan = std::make_unique<LogicalPlan>();
auto boundNode = rel.getSrc()
auto nbrNode = rel.getDst()
auto extendDirection = rel.isUnDirected? BOTH : (boundNode == src ? FWD : BWD);
appendScanNodeID(boundNode, *plan);
appendExtendAndFilter(rel, direction, predicates, *plan);
context->addPlan(newSubgraph, std::move(plan));
// plan dst to src
auto plan = std::make_unique<LogicalPlan>();
auto boundNode = rel.getDst() 
auto nbrNode = rel.getSrc() // swap src and dst
auto extendDirection = rel.isUnDirected? BOTH : (boundNode == src ? FWD : BWD);
appendScanNodeID(boundNode, *plan);
appendExtendAndFilter(rel, direction, predicates, *plan);
context->addPlan(newSubgraph, std::move(plan));
```
Don't forget to abstract common logic of these 2 cases.